src/UI/UIUtils.java

```diff
@@ -0,0 +1,188 @@
+      buildingColorPalettes.put(PURPLE, new ColorPalette(purpleMapBuildingColors));
+
+      mapUnitColorPalettes.put(Color.PINK, new ColorPalette(pinkMapUnitColors));
+      mapUnitColorPalettes.put(Color.CYAN, new ColorPalette(cyanMapUnitColors));
+      mapUnitColorPalettes.put(Color.ORANGE, new ColorPalette(orangeMapUnitColors));
+      mapUnitColorPalettes.put(PURPLE, new ColorPalette(purpleMapUnitColors));
+
+      // Throw some color names in there for the defaults
+      // toString() is not user-friendly
+      colorNames.put(Color.PINK, "salmon");
+      colorNames.put(Color.CYAN, "cyan");
+      colorNames.put(Color.ORANGE, "citrus");
+      colorNames.put(PURPLE, "sparking");
+
+      // We want to be able to use the normal units, as well as any others
+      factions.add(new Faction(DEFAULT_FACTION_NAME,DEFAULT_FACTION_NAME));
+
+      final File folder = new File("res/unit/faction");
+
+      if (folder.canRead())
+      {
+        for( final File fileEntry : folder.listFiles() )
+        {
+          // If it's a file, we assume it's a palette
+          if( !fileEntry.isDirectory() )
+          {
+            String colorName = fileEntry.getAbsolutePath();
+            if( colorName.contains(".png") )
+            {
+              try
+              {
+                BufferedImage bi = ImageIO.read(new File(colorName));
+                // Grab the last color on the first row as our "banner" color
+                Color key = new Color(bi.getRGB(bi.getWidth() - 1, 0));
+                Color[] cUnits = new Color[defaultMapColors.length];
+                Color[] cStructs = new Color[defaultMapColors.length];
+                for( int i = 0; i < defaultMapColors.length; i++ )
+                {
+                  cUnits[i] = new Color(bi.getRGB(i, 0));
+                  cStructs[i] = new Color(bi.getRGB(i, 1));
+                }
+                buildingColorPalettes.put(key, new ColorPalette(cStructs));
+                mapUnitColorPalettes.put(key, new ColorPalette(cUnits));
+                colorNames.put(key, fileEntry.getName().replace(".png", ""));
+              }
+              catch (IOException ioex)
+              {
+                System.out.println("WARNING! Exception loading palette " + colorName);
+              }
+            }
+          }
+          // If it's a directory, we assume it's a set of map sprites, i.e. a faction.
+          else if(fileEntry.canRead() && !fileEntry.getName().endsWith(DEFAULT_FACTION_NAME)) // However, we don't wanna add our default twice
+          {
+            String basis = DEFAULT_FACTION_NAME;
+            try
+            {
+              BufferedReader br = new BufferedReader(new FileReader(fileEntry.getAbsolutePath() + "/basis.txt"));
+              basis = br.readLine().trim();
+              br.close();
+            }
+            catch (IOException ioex)
+            {
+              // if there's no valid basis, there's no problem
```

Comment:
I'd rather check whether the file exists before trying to read, instead of just trying to read and catching an exception if it fails. Exceptions should be for when bad/unexpected/exceptional things happen; it's normal for `basis.txt` to be absent.